yt_dlp/extractor/box.py

```diff
@@ -25,11 +26,34 @@ class BoxIE(InfoExtractor):
             'uploader_id': '235196876',
         },
         'params': {'skip_download': 'dash fragment too small'},
-    }
+    }, {
+        'url': 'https://utexas.app.box.com/s/2x6vanv85fdl8j2eqlcxmv0gp1wvps6e',
+        'info_dict': {
+            'id': '787379022466',
+            'ext': 'mp4',
+            'title': 'Webinar recording: Take the Leap!.mp4',
+            'uploader': 'Patricia Mosele',
+            'timestamp': 1615824864,
+            'upload_date': '20210315',
+            'uploader_id': '239068974',
+        },
+        'params': {'skip_download': 'dash fragment too small'},
+    }]
 
     def _real_extract(self, url):
         shared_name, file_id = self._match_valid_url(url).groups()
-        webpage = self._download_webpage(url, file_id)
+        webpage = self._download_webpage(url, file_id or shared_name)
+
+        if file_id is None:
+            post_stream_data = self._parse_json(self._search_regex(
+                r'Box\.postStreamData\s*=\s*({.+?});', webpage, 'Box post-stream data'),
+                file_id)
+            shared_item = post_stream_data.get('/app-api/enduserapp/shared-item')
+            if shared_item.get('itemType') != 'file':
+                raise ExtractorError('Resource is not a file')
+
+            file_id = str(shared_item.get('itemID'))
```

Comment:
use `traverse_obj` so we don't throw during error handling (which should be `expected=True`, right?)

then let's intentionally KeyError if there's no `itemID`, since we need it for the next request
```suggestion
            shared_item = traverse_obj(post_stream_data, ('/app-api/enduserapp/shared-item', {dict}) or {}
            if shared_item.get('itemType') != 'file':
                raise ExtractorError('The requested resource is not a file', expected=True)

            file_id = str(shared_item['itemID'])
```